barretenberg/cpp/src/barretenberg/commitment_schemes/commitment_key.hpp

```diff
@@ -87,7 +87,12 @@ template <class Curve> class CommitmentKey {
     {
         BB_OP_COUNT_TIME();
         // We must have a power-of-2 SRS points *after* subtracting by start_index.
-        const size_t consumed_srs = numeric::round_up_power_2(polynomial.size()) + polynomial.start_index;
+        // just pick a starting index so that it doesn't go over the dyadic_circuit_size
+        size_t dyadic_poly_size = numeric::round_up_power_2(polynomial.size());
+        size_t relative_start_index =
```

Comment:
can be min(dyadic_poly_size, polynomial.end_index()) - polynomial.size()

Reply:
wasn't sure if that was slower than the ternary operator but probably not

Reply:
lol computation is free, memory access is expensive is a good model to work by. They are equivalent therefore (and indeed compiler emits the same code)

Reply:
(ofc computation being free breaks down when we do bigint math)